.github/workflows/functional-test.yml

```diff
@@ -0,0 +1,54 @@
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Go
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: init directories
+      run: mkdir -p "$(pwd)/bin"
+        
+    - name: Download kubebuilder 
+      run: |
+        version=1.0.8
+        arch=amd64
+        curl -L -O "https://github.com/kubernetes-sigs/kubebuilder/releases/download/v${version}/kubebuilder_${version}_darwin_${arch}.tar.gz"
+        tar -zxvf kubebuilder_${version}_darwin_${arch}.tar.gz
+        mv kubebuilder_${version}_darwin_${arch} kubebuilder
+        echo "kubebuilder_${version}_darwin_${arch} renamed to kubebuilder"
```

Comment:
Hey @abhijeet-dhumal 

I believe this version of kubebuilder, "darwin", is for mac OS. As we are running the tests on `ubuntu-latest`, it maybe necessary to use kubebuilder_linux_amd64

Reply:
Same installation script I used in my local system, it worked!  
Ok, Let me try with `kubebuilder_linux_amd64`